### PR TITLE
Add itertools.islice support in dynamo

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133893

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @xmfan

### CODE REVIEW

**Comment by bobrenjc93** [comment]
This is a resubmission of already approved PR https://github.com/pytorch/pytorch/pull/133820

**Review by oulgen — APPROVED** [review]

File: torch/_dynamo/variables/iter.py [hunk 1]
```diff
@@ -174,6 +174,12 @@ def retrieve_const_key(key):
                     from_exc=e,
                 )
             return variables.ListIteratorVariable(result, mutable_local=MutableLocal())
+        elif self.value is itertools.islice:
+            from .builder import SourcelessBuilder
+
+            return tx.inline_user_function_return(
+                SourcelessBuilder.create(tx, polyfill.islice), args, kwargs
+            )
```
Comment (XuehaiPan) [inline torch/_dynamo/variables/iter.py:182]: I think `islice` is already handled by `BuiltinVariable.call_islice`:

https://github.com/pytorch/pytorch/blob/0d4eacb9d24feca91017439e425953967ebde76e/torch/_dynamo/variables/builtin.py#L1956-L1963

Comment (oulgen) [reply torch/_dynamo/variables/iter.py:182]: It was causing a graph break in the polyfill index code previously, so this approach is more reliable.

**Pull request closed without merging**